packages/capacitor-plugin/ios/Sources/FileTransferPlugin/FileTransferPlugin.swift

```diff
@@ -1,23 +1,236 @@
+            headers: extractHeaders(from: headers),
+            timeout: call.getInt("connectTimeout", 60000) / 1000, // Timeouts in iOS are in seconds. So read the value in millis and divide by 1000
+            disableRedirects: call.getBool("disableRedirects", false),
+            shouldEncodeUrlParams: call.getBool("shouldEncodeUrlParams", true)
+        )
+        
+        return (serverURL, fileURL, shouldTrackProgress, httpOptions)
+    }
+    
+    /// Provides the default HTTP method for the given action.
+    private func defaultHTTPMethod(for action: Action) -> String {
+        switch action {
+        case .download:
+            return "GET"
+        case .upload:
+            return "POST"
+        }
+    }
+    
+    /// Converts a JSObject to a string dictionary used for headers.
+    private func extractHeaders(from jsObject: JSObject) -> [String: String] {
+        var result = [String: String]()
+        for (key, value) in jsObject {
+            result[key] = value as? String ?? ""
+        }
+        return result
+    }
 
-    @objc func echo(_ call: CAPPluginCall) {
-        let value = call.getString("value") ?? ""
-        call.resolve([
-            "value": implementation.echo(value)
-        ])
+    /// Converts a JSObject to a dictionary of arrays, supporting both string and string-array values.
+    private func extractParams(from jsObject: JSObject) -> [String: [String]] {
+        var result: [String: [String]] = [:]
+        for (key, value) in jsObject {
+            if let stringValue = value as? String {
+                result[key] = [stringValue]
+            } else if let arrayValue = value as? [Any] {
+                let stringArray = arrayValue.compactMap { $0 as? String }
+                if !stringArray.isEmpty {
+                    result[key] = stringArray
+                }
+            }
+        }
+        return result
+    }
+    
+    /// Handles completion of the upload or download Combine pipeline.
+    private func handleCompletion(call: CAPPluginCall) -> (Subscribers.Completion<Error>) -> Void {
+        return { completion in
+            if case let .failure(error) = completion {
+                self.sendError(error, call: call)
+            }
+        }
+    }
+
+    /// Handles received value from the Combine stream.
+    ///
+    /// - Parameters:
+    ///   - call: The original plugin call.
+    ///   - type: Whether it's an upload or download.
+    ///   - url: The source or destination URL as string.
+    ///   - path: The file path used in the transfer.
+    ///   - shouldTrackProgress: Whether progress events should be emitted.
+    private func handleReceiveValue(
+        call: CAPPluginCall,
+        type: Action,
+        url: String,
+        path: String,
+        shouldTrackProgress: Bool
+    ) -> (IONFLTRTransferResult) -> Void {
+        return { result in
+            if case let .ongoing(status) = result {
+                if shouldTrackProgress {
```

Comment:
We could use a `guard` instead of this if to avoid nesting, like:

`guard shouldTrackProgress else { return }`